internal/db/sqltest/tests/index/update_time.sql

```diff
@@ -0,0 +1,10 @@
+-- Copyright (c) HashiCorp, Inc.
+-- SPDX-License-Identifier: BUSL-1.1
+
+begin;
+  select plan(1);
+
+  select is(count(*), 32::bigint) from pg_indexes where schemaname = 'public' and indexname like tablename || '_update_time_ix';
```

Comment:
Crazy idea; could we build this test based on the return of `get_deletion_tables()`? It returns all the tables ending in `_deleted`, which should be 1:1 to all tables we want to have an update_time index. Essentially, loop over all the deleted tables, remove the `_deleted` suffix and then call [`has_index`](https://pgtap.org/documentation.html#has_index) on the resulting table name. What do you think? That way, if a user adds a `_deleted` table, we ensure that the original table has an index on the update_time.